File: tests/ClaudeToolbar.Core.Tests/Formatting/BarLevelResolverTests.cs
using ClaudeToolbar.Core.Formatting;

namespace ClaudeToolbar.Core.Tests.Formatting;

public class BarLevelResolverTests
{
    [Theory]
    [InlineData(0, BarLevel.Ok)]
    [InlineData(69.9, BarLevel.Ok)]
    [InlineData(70, BarLevel.Warn)]
    [InlineData(89.9, BarLevel.Warn)]
    [InlineData(90, BarLevel.Crit)]
    [InlineData(100, BarLevel.Crit)]
    public void ResolvesAgainstThresholds(double utilization, BarLevel expected)
    {
        Assert.Equal(expected, BarLevelResolver.Resolve(utilization, 70, 90));
    }
}

File: mac/Tests/ClaudeToolbarCoreTests/Formatting/BarLevelResolverTests.swift
import XCTest
@testable import ClaudeToolbarCore

final class BarLevelResolverTests: XCTestCase {
    func testThresholds() {
        XCTAssertEqual(BarLevelResolver.resolve(utilization: 0, warnThreshold: 70, critThreshold: 90), .ok)
        XCTAssertEqual(BarLevelResolver.resolve(utilization: 69.9, warnThreshold: 70, critThreshold: 90), .ok)
        XCTAssertEqual(BarLevelResolver.resolve(utilization: 70, warnThreshold: 70, critThreshold: 90), .warn)
        XCTAssertEqual(BarLevelResolver.resolve(utilization: 89.9, warnThreshold: 70, critThreshold: 90), .warn)
        XCTAssertEqual(BarLevelResolver.resolve(utilization: 90, warnThreshold: 70, critThreshold: 90), .crit)
        XCTAssertEqual(BarLevelResolver.resolve(utilization: 100, warnThreshold: 70, critThreshold: 90), .crit)
    }
}
